Include equinor tests in test runner and change to pytest mark

oyvindeide opened the pull request:
Equinor tests were not being run in because of a broken symlink,
yet they were marked as passed in ctest. Changed to pytest.mark
to simplify the implementation.

oyvindeide committed:
Include equinor tests in test runner and change to pytest mark

Equinor tests were not being run in because of a broken symlink,
yet they were marked as passed in ctest. Changed to pytest.mark
to simplify the implementation. Jenkins test-running is split
by Equinor/non-Equinor tests.

xjules reviewed (approved):
Tested and worked - good job!

andreabrambilla commented on python/tests/conftest.py:
```diff
@@ -0,0 +1,34 @@
+
+def pytest_runtest_setup(item):
+    if item.get_closest_marker("equinor_test") and not has_equinor_test_data():
+        pytest.skip("Test requires Equinor data")
```
Just a minor thing. Why not replacing this method with something like
```Python
equinor_test = pytest.mark.skipif(lambda: not has_equinor_test_data())
```

This way it is explicit that we are creating a new marker/decorator

oyvindeide replied:
Considered doing something like that, the reason I chose mark was if we wanted the flexibility to just run the Equinor tests with pytest -m "equinor_test", but if that is less readable I'll be happy to change to your suggestion.

oyvindeide committed:
Make gen_kw test run by changing key order

The order of the gen_kw keys have either changed or this test has
never passed. The order is not clear, as the order in the test does
not mirror the order in the config file. If the order matters this
could be a bug. Changed the test to check that the correct keys were
read, and refactored.
